### ISSUE TITLE
ElevationVisualizationConfig.h issue while catkin_make

### ISSUE THREAD

**Issue by suho0515** [tu-darmstadt-ros-pkg/hector_navigation#20]
Hello sir, I'm trying to install hector_navigation package to my pc.
but there is error while compiling.

`[ 62%] Building CXX object hector_navigation/hector_elevation_visualization/CMakeFiles/hector_elevation_visualization_node.dir/src/hector_elevation_visualization_node.cpp.o
In file included from /home/suho/catkin_ws/src/hector_navigation/hector_elevation_visualization/src/hector_elevation_visualization_node.cpp:2:0:
/home/suho/catkin_ws/src/hector_navigation/hector_elevation_visualization/include/hector_elevation_visualization/hector_elevation_visualization.h:12:73: fatal error: hector_elevation_visualization/ElevationVisualizationConfig.h: No such file or directory
compilation terminated.
hector_navigation/hector_elevation_visualization/CMakeFiles/hector_elevation_visualization_node.dir/build.make:62: recipe for target 'hector_navigation/hector_elevation_visualization/CMakeFiles/hector_elevation_visualization_node.dir/src/hector_elevation_visualization_node.cpp.o' failed
make[2]: *** [hector_navigation/hector_elevation_visualization/CMakeFiles/hector_elevation_visualization_node.dir/src/hector_elevation_visualization_node.cpp.o] Error 1
CMakeFiles/Makefile2:3811: recipe for target 'hector_navigation/hector_elevation_visualization/CMakeFiles/hector_elevation_visualization_node.dir/all' failed
make[1]: *** [hector_navigation/hector_elevation_visualization/CMakeFiles/hector_elevation_visualization_node.dir/all] Error 2
Makefile:138: recipe for target 'all' failed
make: *** [all] Error 2
Invoking "make -j1 -l1" failed
`

I think we there should be file named 'ElevationVisualizationConfig.h'.
but i can't find any of them.

is there any way to solve this issue?

**Comment by StefanFabian** [comment]
The header file you're referring to is auto-generated by `dynamic_reconfigure`.
Judging from a quick look there are a couple of small non-critical issues in the `CMakeLists.txt` of that project.
Unfortunately, I currently don't have the time to refactor the packages in this repository but fixing your particular issue should be as simple as adding the following line to the `CMakeLists.txt`:
```
add_dependencies(hector_elevation_visualization_node ${PROJECT_NAME}_gencfg)
```
anywhere after the `add_executable(hector_elevation_visualization_node ...)` line.

**Comment by suho0515** [comment]
> The header file you're referring to is auto-generated by `dynamic_reconfigure`.
> Judging from a quick look there are a couple of small non-critical issues in the `CMakeLists.txt` of that project.
> Unfortunately, I currently don't have the time to refactor the packages in this repository but fixing your particular issue should be as simple as adding the following line to the `CMakeLists.txt`:
> 
> ```
> add_dependencies(hector_elevation_visualization_node ${PROJECT_NAME}_gencfg)
> ```
> 
> anywhere after the `add_executable(hector_elevation_visualization_node ...)` line.

I'd try it later and let you know what happen, thanks @StefanFabian :)

**Comment by ChronostChronz** [comment]
Hello sir, i'm currently using the hector navigation as well and encounter the same issue. Can u help me if it has been done or there's any more solution to be done to make it work? @suho0515 Oh, and i'm using it in Ubuntu Mate 18.04 using ros-melodic

**Comment by Martin-Oehler** [comment]
Have you tested the proposed solution by @StefanFabian?

**Comment by ChronostChronz** [comment]
Yes i added the add_dependancies line and it seems to not working still :( @Martin-Oehler

**Comment by ChronostChronz** [comment]
Nvm it worked on the 2nd try! Thanks a lot!

**Comment by StefanFabian** [comment]
Fixed in 7f95424.
